x/market/types/types.go

```diff
@@ -85,10 +88,17 @@ func (o Order) ValidateCanMatch(height int64) error {
 		return err
 	}
 
-	if o.StartAt > height {
-		return errors.Errorf("too early to match order (%v > %v)", o.StartAt, height)
+	switch {
+	case height > o.StartAt && height < o.CloseAt: // valid to match short circuit
+		return nil
+	case height < o.StartAt:
+		return errors.Wrap(ErrOrderTooEarly, fmt.Sprintf("(%v > %v)", o.StartAt, height))
+	case height > o.CloseAt:
+		// Close Open Order if it have surpassed the CloseAt block height.
+		return ErrOrderDurationExceeded
+	default:
```

Comment:
dude.  no.

if this was about the "return early" comment from a previous revision, i was talking about taking care of simple cases first instead of having a bunch of nesting.

```go
if x == 0 {
  if y > 0 {
    foo()
    return x
  }
  return y
}
return z
```
is better to write as

```go
if x != 0 {
  return z
}
if y == 0 {
  return y
}
foo()
return x
```